docs/hive-integration/kube-api-getting-started.md

```diff
@@ -0,0 +1,493 @@
+# Kube API - Getting Started Guide
+
+This document is a step-by-step guide that demonstrates how to deploy a single-node cluster end to end.
+
+**Note**:
+* This document is not meant to expand on each and every resource in detail; [other documents already do that](README.md). Instead, expect the details needed to understand the context and what is currently happening.
+
+* The order in which actions are performed here is optional, and you may choose to go by a different one. The one order restriction the API does have is in regards to creating NMstateConfigs (which are optional) before InfraEnv, for more details, [check the warning mentioned here](README.md#NMStateConfig).
+## What To Expect
+* A walk through custom resource definitions (CRDs) and understand how they relate to each other.
+
+* A clear understanding of what happened, both in OpenShift / Kubernetes and assisted-service backend per each action described below.
+
+## How To Use This guide
+
+* Make sure that for each step you follow, you understand how and why it is done
+
+* You are encouraged to copy and paste the below-mentioned commands to your terminal and see things in action.
+
+* Note the inline comments in the `yaml` examples. There is no need to remove those in case you which to copy and create the resources directly in your environment.
+
+## Assumptions
+* You have deployed both the operator and assisted-service, [using these instructions](../operator.md).
+
+* You know how to configure BMH. This document use [dev-scripts](https://github.com/openshift-metal3/dev-scripts/) for that. [read about it here](baremetal-agent-controller.md).
+
+
+## Resource Types and Relationships
+![kubeAPI4.9](kubeAPI4.9.jpg)
+
+
+## Cluster Prerequisites
+
+### 1. Create a namespace for your resources
+
+```yaml
+cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: demo-worker4 # Use any name, but not that using 'assisted-installer' is not recommended
+  labels:
+    name: testing-this
+EOF
+```
+
+### 2. Create a Pull Secret
+
+   * Use the secret obtained from [cloud.openshift.com](https://cloud.redhat.com/openshift/install/pull-secret)
+```yaml
+cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/dockerconfigjson
+metadata:
+name: pull-secret
+namespace: demo-worker4
+stringData:
+.dockerconfigjson: '{"auths":{"cloud.openshift.com":{"auth":"your secret here","email":"user@example.com"}}}'
+EOF
+```
+
+### 3. Generate `ClusterImageSet`, to specify the OpenShift version to be used.
+
+```yaml
+cat <<EOF | kubectl create -f -
+apiVersion: hive.openshift.io/v1
+kind: ClusterImageSet
+metadata:
+   name: openshift-v4.8.0
+spec:
+   releaseImage: quay.io/openshift-release-dev/ocp-release:4.8.0-rc.0-x86_64  # That version may change over time
+EOF
+```
+## Creating Cluster Resources
+
+### Register a cluster with assisted-service backend
```

Comment:
The back end is an implementation detail. I would just describe this as defining the cluster.